### ISSUE TITLE
NPE in PagerTabStrip

### ISSUE THREAD

**Issue by dmitriigriazin** [ZieIony/Carbon#130]
My crashlytics shows some fails with PagerTabStrip. They are rare, but pretty stable and deal with different devices. [I'll point the line in the file of master branch.](https://github.com/ZieIony/Carbon/blob/master/carbon/src/main/java/carbon/widget/PagerTabStrip.java#L53)

``` java

private ViewPager.OnPageChangeListener pageChangeListener = new ViewPager.OnPageChangeListener() {
        @Override
        public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
            position = (int) Math.round(position + positionOffset);
            if (position != selectedPage) {
                View view = content.getChildAt(position);
                ...
                animator = ValueAnimator.ofFloat(indicatorPos, view.getLeft());
                ...
                animator2 = ValueAnimator.ofFloat(indicatorPos2, view.getRight());
                ...
```

Sometimes on layout (`carbon.widget.LinearLayout.onLayout (LinearLayout.java:210)`) turns out that getChildAt(pos) returns null. Possibly, you might add null checks for `view`.

Here is a full crash stack if interested:

```
carbon.widget.PagerTabStrip$1.onPageScrolled (PagerTabStrip.java:52)
carbon.widget.ViewPager$1.onPageScrolled (ViewPager.java:38)
android.support.v4.view.ViewPager.onPageScrolled (ViewPager.java:1716)
android.support.v4.view.ViewPager.pageScrolled (ViewPager.java:1637)
android.support.v4.view.ViewPager.scrollToItem (ViewPager.java:585)
android.support.v4.view.ViewPager.onLayout (ViewPager.java:1604)
android.view.View.layout (View.java:13858)
android.view.ViewGroup.layout (ViewGroup.java:4380)
android.widget.LinearLayout.setChildFrame (LinearLayout.java:1649)
android.widget.LinearLayout.layoutVertical (LinearLayout.java:1507)
android.widget.LinearLayout.onLayout (LinearLayout.java:1420)
carbon.widget.LinearLayout.onLayout (LinearLayout.java:210)
android.view.View.layout (View.java:13858)
android.view.ViewGroup.layout (ViewGroup.java:4380)
android.widget.RelativeLayout.onLayout (RelativeLayout.java:948)
carbon.widget.RelativeLayout.onLayout (RelativeLayout.java:211)
android.view.View.layout (View.java:13858)
android.view.ViewGroup.layout (ViewGroup.java:4380)
android.widget.FrameLayout.onLayout (FrameLayout.java:448)
android.view.View.layout (View.java:13858)
android.view.ViewGroup.layout (ViewGroup.java:4380)
android.widget.RelativeLayout.onLayout (RelativeLayout.java:948)
android.view.View.layout (View.java:13858)
android.view.ViewGroup.layout (ViewGroup.java:4380)
android.support.v4.widget.DrawerLayout.onLayout (DrawerLayout.java:907)
android.view.View.layout (View.java:13858)
android.view.ViewGroup.layout (ViewGroup.java:4380)
android.widget.FrameLayout.onLayout (FrameLayout.java:448)
android.view.View.layout (View.java:13858)
android.view.ViewGroup.layout (ViewGroup.java:4380)
android.widget.LinearLayout.setChildFrame (LinearLayout.java:1649)
android.widget.LinearLayout.layoutVertical (LinearLayout.java:1507)
android.widget.LinearLayout.onLayout (LinearLayout.java:1420)
android.view.View.layout (View.java:13858)
android.view.ViewGroup.layout (ViewGroup.java:4380)
android.widget.FrameLayout.onLayout (FrameLayout.java:448)
android.view.View.layout (View.java:13858)
android.view.ViewGroup.layout (ViewGroup.java:4380)
android.widget.LinearLayout.setChildFrame (LinearLayout.java:1649)
android.widget.LinearLayout.layoutVertical (LinearLayout.java:1507)
android.widget.LinearLayout.onLayout (LinearLayout.java:1420)
android.view.View.layout (View.java:13858)
android.view.ViewGroup.layout (ViewGroup.java:4380)
android.widget.FrameLayout.onLayout (FrameLayout.java:448)
android.view.View.layout (View.java:13858)
android.view.ViewGroup.layout (ViewGroup.java:4380)
android.view.ViewRootImpl.performLayout (ViewRootImpl.java:2106)
android.view.ViewRootImpl.performTraversals (ViewRootImpl.java:1907)
android.view.ViewRootImpl.doTraversal (ViewRootImpl.java:1148)
android.view.ViewRootImpl$TraversalRunnable.run (ViewRootImpl.java:4656)
android.view.Choreographer$CallbackRecord.run (Choreographer.java:734)
android.view.Choreographer.doCallbacks (Choreographer.java:560)
android.view.Choreographer.doFrame (Choreographer.java:529)
android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:719)
```

**Comment by ZieIony** [comment]
I totally forgot about this issue. Seems like it's happening durning state restore if there's no child PagerTabStrip was scrolled to before save. I've added a check, so it should be fine now, but I'm curious if can be done better.

**Comment by dmitriigriazin** [comment]
hi @ZieIony 

I was moved to tears. It's been so long, like 5 years have passed. 
I just got married. I raised two children since then. So much has happened. So one can even think those issues do not matter anymore.
 
Until I accidentally come to the old issues and see them fixed. No need to keep a private fork, I can just use deps

Gosh, it's such a great feeling. Relief. 

Bugfixing matters. Tnx bro
